enhancements/scorecard-storage.md

```diff
@@ -0,0 +1,264 @@
+
+QE teams for example might already have tests which produce test output files, in order for them to integrate their tests into scorecard, they need a persistent volume to write test results into.
+
+This feature enables a wider adoption of tests that can be executed within scorecard.
+
+### Goals
+
+Goals of this proposal include:
+* provide a persistent volume for scorecard test developers
+* Collect persistent volume output when scorecard completes
+
+### Non-Goals
+
+This enhancement doesn't change scorecard's test result format (v1alpha3) so this is not a breaking change.
+
+## Proposal
+
+### User Stories
+
+#### Story 1
+A scorecard user can specify for a given test that they want persistent storage to be made available to that test.  This is specified within the scorecard configuration file as a label.  For example, a user would specify they want storage for a test by specifying
+in the scorecard configuration file the `storage` label as in the following example:
+
+```yaml
+  tests:
+  - image: quay.io/custom-scorecard-example/example-test:latest
+    entrypoint:
+    - example-scorecard-test
+    - example-test
+    labels:
+      suite: example
+      test: example-test
+      storage: true
+```
+
+#### Story 2
+A scorecard user can optionally specify storage class details within the scorecard configuration file to override storage defaults. Volume size, access mode, and storage class names can be specified as follows:
+```yaml
+  tests:
+  - image: quay.io/custom-scorecard-example/example-test:latest
+    storage: 
+      provision: true
+      size: 1Gi
+      class: standard
+    entrypoint:
+    - example-scorecard-test
+    - example-test
+    labels:
+      suite: custom
+      test: example-test
+      storage: true
+      storage-size: 1Gi
+      storage-accessmode: ReadWriteOnce
+      storage-class: hostpath
+```
+
+Valid values for access mode:
+ * ReadWriteOnce (default if not specified)
+ * ReadOnlyMany
+ * ReadWriteMany
+
+Access mode values are defined in https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/api/core/v1/types.go#L556
+
+Valid values for volume size must be formatted according to the rules defined for storage 
+in https://pkg.go.dev/k8s.io/apimachinery/pkg/api/resource#Format.  Typical values for 
+storage are like:
+ * 1Gi (1 gigabyte) (default)
+ * 100M (100 megabyte)
+
+Storage classes can be found on a cluster by the following command:
+ * kubectl get sc
+
+A default storage class is the storage class on a cluster that has the following annotation:
+```yaml
+  metadata:
+    annotations:
+      storageclass.kubernetes.io/is-default-class: "true"
+```
+
+
+#### Story 3
+Scorecard gathers any persisted volume data upon test completion and stores it in a local directory where scorecard is executed.  Users
+can specify a non-default local path for test output using a command flag as follows:
+```bash
+operator-sdk scorecard ./bundle --selector=suite=custom --test-output=/mytestoutput
```

Comment:
the actual output from scorecard+the tests shows up on stdout?

Reply:
yes, today scorecard output showing pass/fail just goes to stdout.